### PR TITLE
Cache timestamp to avoid compare_func from crashing

### PR DESCRIPTION
**Pull request by dbhowell**
Avoids using freed nodes when `ConversationListStore` uses it's `compare_func`.

Fixed the `camel_message_info_get_date_received: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed` errors for me (#649, #650 , #652).

Note: `folder_changed` updates *every* conversation in the folder, so is expensive when a folder has a large number of items. So for example, setting a message as read will ultimately mean `folder_changed` fires and touches every conversation. I can't see a way around it at the moment though.

### CODE REVIEW

**Comment by marbetschar** [comment]
Neat, thank you @dbhowell!! @RMcNeely / @wout / @casasfernando: Can anyone of you guys confirm that this fix prevents Mail from crashing for you as well?

**Comment by casasfernando** [comment]
I will try it out during today and report back.

**Comment by davidmhewitt** [comment]
I've just pushed another commit to #663 that might help narrow down the root cause of this too.

If possible, could you test both?

**Comment by casasfernando** [comment]
Sure. I can test both. :)

**Comment by casasfernando** [comment]
> I've just pushed another commit to #663 that might help narrow down the root cause of this too.
> 
> If possible, could you test both?

@davidmhewitt I just updated #650 with a new trace after pulling the latest commits from your branch.

Thanks

**Comment by casasfernando** [comment]
Congrats @dbhowell your patch fixed the crash on startup for me as well!
I will test it a bit more but so far so good.

Thanks.

File: src/ConversationList/ConversationItemModel.vala [hunk 1]
```diff
@@ -153,7 +155,7 @@ public class Mail.ConversationItemModel : GLib.Object {
 
     public int64 timestamp {
```
Comment (tintou) [inline src/ConversationList/ConversationItemModel.vala:156]:
```suggestion
    public int64 timestamp { get; private set; }
```
and then simply use `timestamp = get_newest_timestamp (new_node, -1);`

**Comment by tintou** [comment]
It looks great, just adding a comment to have a more concise code but it's indeed avoiding issues

**Review by tintou — APPROVED** [review]

**Pull request merged**